### PR TITLE
feat: convenience api for flags assets

### PR DESCRIPTION
**Pull request by florentlotthepro**
## Changes description

Add an API to get flags assets from a string.

## Context

When we need to use a flag, we have a country code but currently, with the country code, we cannot get the flag.

## Checklist

- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on a device/simulator.

## Does this introduce a breaking change?

- No

## Other information

How to use:
```swift
let frenchFlag: UIImage? = VitaminAssets.flag(countryCode: "fr")?.image
```

### CODE REVIEW

File: Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//
+//  Vitamin iOS
+//  Apache License 2.0
+//
+
+import Foundation
+
+extension VitaminAssets {
+    /// Return the country flag using the country code.
+    /// - Parameter countryCode: alpha-2 country code (ISO 3166-1).
```
Comment (daniel-dumortier) [inline Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift:10]: Some of the flag assets have a three-character name (`CAF`, `CAS`, `CEU`, `CNA`, `COC`, `CSA`, `DOM`, `IND`, `ISL`)
Should we rename the assets (with a potential breaking change) ?
Should we have a correspondance table between actual codes and assets names ?
Should we just document these specific cases ?

**Review by daniel-dumortier — COMMENTED** [review]
OK for me technically speaking, but in some "exotic" cases, the documentation is wrong

File: Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift (hunk 1 above)
Comment (florentlotthepro) [reply Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift:10]: Well seen! I think that it's not normal to have 2 different kinds of country codes 😕 I'll use the alpha-2 country code everywhere but without breaking changes (just some deprecated notices).
There are also some no country flags, so for these, I think that we can just ignore them and use them like before. 
It's good for you?

Comment (daniel-dumortier) [reply Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift:10]: Ok for me, but VitaminAssets.swift is a SwiftGen generated class... So I think we should avoid tampering with it.
How will you handle the deprecation ? With an VitaminAssets extension where you will hardcode the old three-character country codes (DOM, IND, ISL) ?
If yes, that's seems me ok.

And ok for the other "exotic" sub country flags that will be accessible the old way, or by forcing a non ISO code in the new API

Comment (florentlotthepro) [reply Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift:10]: I just update the PR, I made an extension outside of the SwiftGen class to make a link to the new image name.

File: Sources/VitaminSwiftUI/Foundations/Assets/README.md [hunk 2]
```diff
@@ -21,6 +21,12 @@ struct AssetsView: View {
 }
 ```
 
+
+For the flags, you can also use the specific API with alpha-2 country code (ISO 3166-1).
+```swift
+let frenchFlag: UIImage? = VitaminAssets.flag(countryCode: "fr")?.swiftUIImage
```
Comment (daniel-dumortier) [inline Sources/VitaminSwiftUI/Foundations/Assets/README.md:27]: I think there is an error due to a too quick copy/paste from UIKit README, no ? :)

```suggestion
let frenchFlag: Image? = VitaminAssets.flag(countryCode: "fr")?.swiftUIImage
```

File: Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift [hunk 3]
```diff
@@ -13,3 +13,12 @@ extension VitaminAssets {
         VitaminAsset(name: countryCode.uppercased())
     }
 }
```
Comment (daniel-dumortier) [inline Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift:15]: I don't know where to write it, so I write it here : shouldn't the file be named `VitaminAssets+Flags.swift` to respect the convention we applied for extensions on system types ?
exemple : `UIImage+SwiftUI.swift` or `UIImage+Color.swift`

Really not blocking, you can leave it as is if you prefer, I just wanted to ask for potential future extension files.

**Review by daniel-dumortier — CHANGES_REQUESTED** [review]
Just two little remarks on the SwiftUI doc and on extension file naming.
And everything will be ok for me

File: Sources/VitaminSwiftUI/Foundations/Assets/README.md (hunk 2 above)
Comment (florentlotthepro) [reply Sources/VitaminSwiftUI/Foundations/Assets/README.md:27]: Sorry for that 🙈

File: Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift (hunk 3 above)
Comment (florentlotthepro) [reply Sources/VitaminCore/Foundations/Assets/VitaminAssetsFlags.swift:15]: Yes, you're right, sorry for that too. I'll change it.

**Review by daniel-dumortier — APPROVED** [review]
Good job ! Everything is fine for me now.

**Pull request merged**